include/vast/Translation/CodeGenContext.hpp

```diff
@@ -198,13 +201,40 @@ namespace vast::cg
             return symbol(methdecls, mangled, "undeclared method '" + mangled.name + "'", with_error);
         }
 
+        hl::DtorOp lookup_destructor(mangled_name_ref mangled, bool with_error = true) {
+            return symbol(dtordecls, mangled, "undeclared destructor '" + mangled.name + "'", with_error);
+        }
+
+        template< typename Op >
+        struct is_funclike {
```

Comment:
I am pretty sure there is some `mlir::` interface for this. Is there a reason it cannot be used?

Also we usually use `type_list` for these types of queries.